bookwyrm/models/report.py

```diff
@@ -22,28 +24,61 @@
 DELETE_ITEM = "delete_item"
 
 
-class Report(BookWyrmModel):
+class Report(ActivityMixin, BookWyrmModel):
     """reported status or user"""
 
-    reporter = models.ForeignKey(
-        "User", related_name="reporter", on_delete=models.PROTECT
+    activity_serializer = activitypub.Flag
+
+    user = fields.ForeignKey(
+        "User",
+        on_delete=models.PROTECT,
+        activitypub_field="actor",
     )
-    note = models.TextField(null=True, blank=True)
-    user = models.ForeignKey("User", on_delete=models.PROTECT, null=True, blank=True)
-    status = models.ForeignKey(
+    note = fields.TextField(null=True, blank=True, activitypub_field="content")
+    reported_user = fields.ForeignKey(
+        "User",
+        related_name="reported_user",
+        on_delete=models.PROTECT,
+        null=True,
+        blank=True,
+        activitypub_field="to",
+    )
+    statuses = models.ManyToManyField(
         "Status",
         null=True,
         blank=True,
-        on_delete=models.PROTECT,
     )
-    links = models.ManyToManyField("Link", blank=True)
+    links = fields.ManyToManyField("Link", blank=True)
     resolved = models.BooleanField(default=False)
-
-    def raise_not_editable(self, viewer):
-        """instead of user being the owner field, it's reporter"""
-        if self.reporter == viewer or viewer.has_perm("bookwyrm.moderate_user"):
+    allow_broadcast = models.BooleanField(default=False)
+
+    def to_activity_dataclass(self, **kwargs):
+        """generate the objects field"""
+        activity = super().to_activity_dataclass(**kwargs)
+        activity.object = self.object()
+        return activity
+
+    def object(self):
+        """Generate a list of reported objects in a format Mastodon will like"""
+        items = [self.reported_user.remote_id]
+        if self.statuses:
```

Comment:
When I test this `self.statuses` is always `bookwyrm.Status.None` here. So the status is never sent to Mastodon. _However_ `statuses` obviously _isn't_ empty because in the BookWyrm admin interface I can see it. So I have no idea what is going on but it's weird and means statuses aren't passed through to federated servers with the report.